extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -17,43 +17,61 @@ import * as Preact from '../../../src/preact';
 import {Arrow} from './arrow';
 import {Scroller} from './scroller';
 import {toChildArray, useRef, useState} from '../../../src/preact';
+import {useMountEffect} from '../../../src/preact/utils';
 
 /**
- * @param {!JsonObject} props
+ * @param {!BaseCarouselDef.Props} props
  * @return {PreactDef.Renderable}
  */
 export function BaseCarousel(props) {
   const {
-    'arrowPrev': arrowPrev,
-    'arrowNext': arrowNext,
-    'children': children,
-    'defaultSlide': defaultSlide,
-    'loop': loop,
+    arrowPrev,
+    arrowNext,
+    children,
+    defaultSlide,
+    loop,
+    slide,
+    onSlideChange,
+    onMount,
+    setAdvance,
     ...rest
   } = props;
   const childrenArray = toChildArray(children);
   const {length} = childrenArray;
   const [curSlide, setCurSlide] = useState(defaultSlide || 0);
+  const slideState = slide ? slide : curSlide;
```

Comment:
This will have a bug if `slide` is `0`. Try:
```suggestion
  const slideState = slide ?? curSlide;
```